proto/types/types.proto

```diff
@@ -169,3 +170,17 @@ message SignedHeader {
   Header header = 1;
   Commit commit = 2;
 }
+
+message ValidatorSet {
+  repeated Validator validators         = 1;
+  Validator          proposer           = 2;
+  int64              total_voting_power = 3;  // TODO: change to uint64
+}
+
+message Validator {
+  bytes                             address     = 1;
+  tendermint.proto.crypto.PublicKey pub_key     = 2;
```

Comment:
q: why not use bytes directly?

Reply:
with the current Tendermint I could get away with this since we use only one type of key but if we wanted more keys we wouldn't know the type of key that is being sent over the wire by the only bytes. This mainly enables future modularity